test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -4768,6 +4768,693 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
+                var folderNuGetProject = msBuildNuGetProject.FolderNuGetProject;
+                var packageIdentity = MorePackageWithDependents[3]; // Microsoft.Net.Http.2.2.22
+
+                await nuGetPackageManager.InstallPackageAsync(msBuildNuGetProject, packageIdentity,
+                    new ResolutionContext(), new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories().First(), null, token);
+
+                var resolutionContext = new ResolutionContext(
+                    DependencyBehavior.Highest,
+                    false,
+                    true,
+                    VersionConstraints.ExactMajor | VersionConstraints.ExactMinor | VersionConstraints.ExactPatch |
+                    VersionConstraints.ExactRelease);
+
+                // Act
+                var packageActions = (await nuGetPackageManager.PreviewUpdatePackagesAsync(
+                    msBuildNuGetProject,
+                    resolutionContext,
+                    new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories(),
+                    sourceRepositoryProvider.GetRepositories(),
+                    token)).ToList();
+
+                // Assert
+                var singlePackageSource = sourceRepositoryProvider.GetRepositories().Single().PackageSource.Source;
+                Assert.Equal(6, packageActions.Count);
+
+                var batchStartId = string.Empty;
+                var batchEndId = string.Empty;
+                var projectName = string.Empty;
+
+                // add batch events handler
+                nuGetPackageManager.BatchStart += (o, args) =>
+                {
+                    batchStartId = args.Id;
+                    projectName = args.Name;
+                };
+
+                nuGetPackageManager.BatchEnd += (o, args) =>
+                {
+                    batchEndId = args.Id;
+                };
+
+
+                // Main Act
+                await
+                    nuGetPackageManager.ExecuteNuGetProjectActionsAsync(msBuildNuGetProject, packageActions,
+                        new TestNuGetProjectContext(), token);
+
+                // Check that the packages.config file exists after the installation
+                Assert.True(File.Exists(packagesConfigPath));
+                // Check the number of packages and packages returned by PackagesConfigProject after the installation
+                var packagesInPackagesConfig =
+                    (await msBuildNuGetProject.PackagesConfigNuGetProject.GetInstalledPackagesAsync(token)).ToList();
+                Assert.Equal(3, packagesInPackagesConfig.Count);
+                Assert.True(File.Exists(folderNuGetProject.GetInstalledPackageFilePath(packageIdentity)));
+                Assert.NotEmpty(batchStartId);
+                Assert.NotEmpty(batchEndId);
+                Assert.Equal(batchStartId, batchEndId);
```

Comment:
This test should verify that the batch event is only raised once, but here it could be raised multiple times and this would still be true. Add a count here.